File: MAP/Lab 2/ToyInterpreter/src/view/commands/RunAllStepsCommand.java
package view.commands;

import controller.Controller;
import model.state.PrgState;

import java.util.List;
import java.util.Scanner;

public class RunAllStepsCommand extends Command {

    public RunAllStepsCommand(String key, String description, Controller controller) {
        super(key, description, controller);
    }

    @Override
    public void execute() throws Exception {
        executeAllSteps();
        printProgram();
    }

    private int getNr(String msg) {
        System.out.print(msg);

        Scanner sc = new Scanner(System.in);
        return sc.nextInt();
    }

    private void executeOneStep() throws Exception {
        controller.oneStepForAll((List<PrgState>) controller.getPrograms().getData());
    }

    private void printProgram() {
        System.out.println( controller.getPrograms().toString()  );
    }

    private void executeAllSteps() throws Exception {
        controller.allStep();
    }
}

File: MAP/Lab10/ConsoleApp1/ConsoleApp1/View/commands/RunAllStepsCommand.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ToyInterpreter.View.commands
{
    class RunAllStepsCommand: Command
    {
        public RunAllStepsCommand(String key, String description, Controller.Controller controller): base(key, description, controller)
        {
      
        }


        public override void execute()
        {
            executeAllSteps();
            printProgram();
        }

        private int getNr(String msg)
        {
            Console.WriteLine(msg);

            return Convert.ToInt32(Console.ReadLine());
        }

        private void executeOneStep()
        {
            controller.oneStep(controller.getFirstProgram());
   
        }

        private void printProgram()
        {
            Console.WriteLine(controller.ToString());
        }

        private void executeAllSteps()
        {
            controller.allSteps();
        }
}
}
